client/components/domains/register-domain-step/index.jsx

```diff
@@ -541,10 +539,20 @@ const RegisterDomainStep = React.createClass( {
 				severity = 'info';
 				break;
 			case 'not_registrable':
-				if ( domain.indexOf( '.' ) ) {
-					message = this.translate( 'Sorry but %(domain)s cannot be registered on WordPress.com.', {
-						args: { domain: domain }
-					} );
+				const tldIndex = domain.lastIndexOf( '.' );
```

Comment:
There's a `isSubdomain` function in https://github.com/Automattic/wp-calypso/blob/master/client/lib/domains/index.js#L114 which might be useful